### PR TITLE
Add code promotion use case to GitFS Walkthrough

### PR DESCRIPTION
**Pull request by terminalmage**
### What does this PR do?

Adds a use case to the [GitFS Walkthrough](https://docs.saltproject.io/en/latest/topics/tutorials/gitfs.html) which explains how to use gitfs to achieve a code promotion workflow.

### What issues does this PR fix or reference?
Refs: https://github.com/saltstack/salt/issues/46660

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
- [X] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] ~Tests written/updated~ **N/A**

### CODE REVIEW

File: doc/topics/tutorials/gitfs.rst [hunk 1]
```diff
@@ -796,6 +791,113 @@ To map a branch other than ``master`` as the ``base`` environment, use the
 The base can also be configured on a :ref:`per-remote basis
 <gitfs-per-remote-config>`.
 
+Use Case: Code Promotion (dev -> qa -> base)
+--------------------------------------------
+
+When running a :ref:`highstate <running-highstate>`, the ``top.sls`` files from
+different from all of the branches and tags will be merged into one. This does
```
Comment (Ch3LL) [inline doc/topics/tutorials/gitfs.rst:798]: This wording seems a bit off. "from different from"

File: doc/topics/tutorials/gitfs.rst [hunk 2]
```diff
@@ -796,6 +791,113 @@ To map a branch other than ``master`` as the ``base`` environment, use the
...
+     minion config file, and pairs nicely with cases where you do not have
+     separate salt deployments for dev/qa/prod. You can have a box with
+     :conf_minion:`saltenv` set to ``base``, which you can test your dev
+     changes on by running your ``state.apply`` with ``saltenv=dev``.
+
+   - If you don't set :conf_minion:`saltenv` in the minion config file, you
+     _must_ specify it at runtime to avoid conflicting IDs.
+
+
+If you branched ``qa`` of of ``master``, and ``dev`` off of ``qa``, you can
```
Comment (Ch3LL) [inline doc/topics/tutorials/gitfs.rst:897]:
```suggestion
If you branched ``qa`` off of ``master``, and ``dev`` off of ``qa``, you can
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: doc/topics/tutorials/gitfs.rst (hunk 1 above)
Comment (terminalmage) [reply doc/topics/tutorials/gitfs.rst]: Yeah, not what I originally wanted to type here, this is fixed.

File: doc/topics/tutorials/gitfs.rst (hunk 2 above)
Comment (terminalmage) [reply doc/topics/tutorials/gitfs.rst]: Good catch, also fixed.

**Comment by terminalmage** [comment]
@Ch3LL this is ready for re-review

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**